data/state/peerAccount.go

```diff
@@ -368,6 +371,12 @@ func (pa *PeerAccount) GetRating() uint32 {
 	return pa.Rating
 }
 
+// SetListAndIndex will update the peer's list (eligible, waiting) and the index inside it
+func (pa *PeerAccount) SetListAndIndex(list string, index int) {
+	pa.List = list
```

Comment:
I think you should save with journal the changes

Reply:
done